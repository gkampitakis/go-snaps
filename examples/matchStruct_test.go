package examples_test

import (
	"testing"
	"time"

	"github.com/gkampitakis/go-snaps/match"
	"github.com/gkampitakis/go-snaps/snaps"
)

type User struct {
	Name     string
	Age      int
	internal string
	Data     map[string]int
	Nested   struct {
		Metadata string
		Date     time.Time
	}
}

func TestMatchStruct(t *testing.T) {
	user := User{Name: "george", Age: 10, internal: "metadatav2",
		Nested: struct {
			Metadata string
			Date     time.Time
		}{
			Metadata: "some more data here",
			Date:     time.Now(),
		}}

	t.Run("should create a simple snapshot", func(t *testing.T) {
		//  "internal"
		snaps.MatchStruct(t, &user, match.Any("Nested").ErrOnMissingPath(false))
	})
}
