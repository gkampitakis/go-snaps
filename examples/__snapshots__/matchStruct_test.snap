
[TestMatchStruct/should_create_a_simple_snapshot - 1]
examples_test.User{
    Name:     "george",
    Age:      10,
    internal: "metadatav2",
    Nested:   struct { Metadata string; Date time.Time }{
        Metadata: "some more data here",
        Date:     time.Date(2023, time.May, 29, 22, 23, 19, 588925000, time.Local),
    },
}
---
